js/pages/settings.js

```diff
@@ -1,327 +1,996 @@
-import { showAnimation, hideAnimation, getMyData, storeResponse, getConceptVariableName } from "../shared.js";
+import { allStates, showAnimation, hideAnimation, getMyData } from '../shared.js';
+import { changeEmail, changeContactInformation, changeMailingAddress, changeName, FormTypes, getCheckedRadioButtonValue, handleContactInformationRadioButtonPresets, handleOptionalFieldVisibility, hideEditButtonsUntilUserVerified, hideSuccessMessage, showAndPushElementToArrayIfExists, suffixList, suffixToTextMap, toggleElementVisibility, togglePendingVerificationMessage, updatePhoneNumberInputFocus, validateContactInformation, validateEmailAddress, validateMailingAddress, validateName } from '../settingsHelpers.js';
+import { addEventAddressAutoComplete } from '../event.js';
+import cId from '../fieldToConceptIdMapping.js';
+
+let middleName;
+let suffix;
+let preferredFirstName;
+let otherPhoneNumberComplete;
+let additionalEmail;
 
 export const renderSettingsPage = async () => {
-    document.title = 'My Connect - My Profile';
-    showAnimation();
-    const myData = await getMyData();
-    let template = ''
-    
-    if(myData.code === 200 && myData.data['699625233'] !== undefined && myData.data['699625233'] === 353358909){
-        const userData = myData.data;
-
-        template += `
+  document.title = 'My Connect - My Profile';
+  showAnimation();
+  const myData = await getMyData();
+  const userData = myData.data;
```

Comment:
Should we check the data exists

Reply:
On one hand the user doc is written by cloud functions and the user is authenticated at this point in the app flow. Chances are almost nonexistent that there won't be a user doc. On the other hand chances are probably not zero - it seems like a worthwhile check - I agree with you and I'll add this. Thanks 👍